File: SlidingWindow/BuySellStocks/StockMarket.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left = 0
        right = 1
        maxProfit = 0

        while (right < len(prices)):
            #means found one even less, make left to that to check
            if (prices[right] < prices[left]):
                left = right
                right = left + 1
            #means right is bigger than left, so potential profit
            else:
                profit = prices[right] - prices[left]
                if (profit > maxProfit):
                    maxProfit = profit
                right += 1
        return maxProfit

File: SlidingWindow/BuySellStocks/StockMarket.cpp
class Solution {
public:
/*
- story (of problem in terms of what should do): find max difference between two numbers
- input: array      
- output: max profit (positive difference) or 0 if no positive difference exists
- Requirements: IMPORTANT ESP EDGE CASES
- Constraints that could help:
- 

- Algo (thinking of): sliding window (max and left and right)
*/
    int maxProfit(vector<int>& prices) {

        //since starts at 0, will return 0 if not profit
        int maxProfit = 0;
        size_t left = 0;
        size_t right = 1;

        while (right < prices.size())
        {
            int profit = prices[right]-prices[left];
            if (prices[left] > prices[right])
            {
                left = right;
            }
            else if (profit > maxProfit)
            {
                maxProfit = profit;
            }
            right++;
        }

        return maxProfit;
    }
};
